core/base_algo.py

```diff
@@ -372,20 +372,23 @@ def get_remaining_categories(self) -> Set:
         return ALL_CATEGORIES - categories_in_file
 
     def _update_results_csv(self, values_dict) -> None:
-        key = [self.args.category, self.model_name]
-        if self.experiment_results_manager.data[['category', 'model_name']].isin(key).any(axis=1).any():
-            # Key match found, overwriting
-            mask = (self.experiment_results_manager.data[['category', 'model_name']] != key)
-            self.experiment_results_manager.data = self.experiment_results_manager.data[mask]
-        
-        new_dict = values_dict.copy()
-        new_dict['category'] = self.args.category
-        new_dict['category_type'] = CATEGORY_TO_TYPE[self.args.category]
-        new_dict['model_name'] = self.model_name
-        self.experiment_results_manager.data = \
-            pd.concat(
-                [self.experiment_results_manager.data, pd.DataFrame(new_dict, index=[0])]
-            ).reset_index(drop=True).sort_values(['model_name', 'category_type', 'category'])
+        with self.experiment_results_manager.lock:
```

Comment:
if the classname isn't DataFrameManager then maybe we need to change the name of the data member to `self.persistent_results_df`

Reply:
I think its fine because it is still the experiment result manager, it just happens to be an instance of the class "PersistentDF", but it could very well just be a simple dataframe